htslib/sam.h

```diff
@@ -1355,6 +1356,188 @@ const char *sam_parse_region(sam_hdr_t *h, const char *s, int *tid,
      *** Manipulating auxiliary fields ***
      *************************************/
 
+/// Converts a BAM aux tag to SAM format
+/*
+ * @param b   Pointer to the bam record
+ * @param tag Tag pointer, as returned by bam_aux_get.
+ * @param end Pointer to end of bam record (largest extent of tag).
+ * @param ks  Kstring to write the formatted tag to.
+ *
+ * @return pointer to end of tag on success,
+ *         NULL on failure.
+ */
+static inline uint8_t *sam_format_aux1(const bam1_t *b,
+                                       uint8_t *key, uint8_t *tag,
```

Comment:
Actually I notice that `b` isn't even used!  I assume it was at some point and I removed it during development.

So that begs the question whether this should be an arbitrary "binary to text" conversion from any source, in which case `b` should be removed and `end` kept, or whether it should be tied specifically to a BAM record in which case as you say `end` can be computed.